### ISSUE TITLE
Comment on lab 2

### ISSUE THREAD

**Issue by mattiaottoborgo** [MicheleCazzola/CI2024_lab2#1]
First of all, I want to say that your solution is really good! Clean and understandable implementation. The ReadME helped me a lot to understand your implementation.
I would like to make the following observation/suggestions to help to tackle the difficulties you pointed:
You mention that for the italian instance your algorithm seems to get stucked in a good solution and doesn't manage to further improve it. I think the problem might be diversity:
Your initialization technique is really good since it allows the algorithm to start from a solid base, however I think it might lack of diversity. If you think about it, the initial population, which is composed of individuals generated with greedy algorithm, will likely be composed of really similar individuals. I think you could try to implement some mechanism that generates just some individuals based on your greedy algorithm and the rest in a randomic way. I observed you have a similar setup when the initial population is less than the population size. This might be a good start, however it doesn't guarantee that, when the randomic individuals are far less than the greedy ones, they might survive enough to produce interesting results. To solve this, I would also implement some strategy such as "islands" to preserve such diversity.
In conclusion:
-Really good algorithm to start with
- Try to add more random individuals and choose the strategy you prefer to preserve diversity!
